### PR TITLE
Custom logging Elasticseach image

### PR DESCRIPTION
**Pull request by corvus-ch**
The default variables of the `openshift_logging_elasticsearch` role suggest that the image pull secret can be overridden only for Elasticsearch. It actually was only possible to do so for all logging components. This commit actually makes use of `openshift_logging_elasticsearch_image_pull_secret` which by default gets assigned the broader `openshift_hosted_logging_image_pull_secret` or `openshift_logging_image_pull_secret` values.

### CODE REVIEW

File: playbooks/init/base_packages.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@
       - "{{ 'python-ipaddress' if ansible_distribution != 'Fedora' else '' }}"
       - libsemanage-python
       - yum-utils
-      - "{{ 'python3-docker' if ansible_distribution == 'Fedora' else 'python-docker' }}"
+      - "{{ 'python3-docker' if ansible_distribution == 'Fedora' else 'python-docker-py' }}"
```
Comment (corvus-ch) [inline playbooks/init/base_packages.yml]: Actually I am unsure about this one. Looks a lot like scope creep but was a necessary change to get my test environment working.

File: roles/openshift_logging_elasticsearch/defaults/main.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ---
 ### Common settings
-openshift_logging_elasticsearch_image_pull_secret: "{{ openshift_hosted_logging_image_pull_secret | default('') }}"
+openshift_logging_elasticsearch_image_pull_secret: "{{ openshift_hosted_logging_image_pull_secret | default(openshift_logging_image_pull_secret) }}"
```
Comment (mhutter) [inline roles/openshift_logging_elasticsearch/defaults/main.yml:3]: What happens if `openshift_logging_image_pull_secret` is not set? Is `default('')` required?

File: playbooks/init/base_packages.yml (hunk 1 above)
Comment (mhutter) [reply playbooks/init/base_packages.yml]: If it's intended to send a PR to upstream this should probably be its own PR, otherwise I don't care

File: roles/openshift_logging_elasticsearch/defaults/main.yml (hunk 2 above)
Comment (corvus-ch) [reply roles/openshift_logging_elasticsearch/defaults/main.yml:3]: No it is not `openshift_logging_image_pull_secret` is set as default variable in the role `openshift_logging`.

File: playbooks/init/base_packages.yml (hunk 1 above)
Comment (corvus-ch) [reply playbooks/init/base_packages.yml]: Actually once I rebased to upstream, this changed vanished. It was already changed there but we did not pull yet. I will remove it from this PR.
